packages/tenant-process/src/services/tenantService.ts

```diff
@@ -669,37 +758,19 @@ async function assignCertifiedAttribute({
 
 async function reevaluateTenantKind({
   tenant,
-  version,
   readModelService,
-  correlationId,
 }: {
   tenant: Tenant;
-  version: number;
   readModelService: ReadModelService;
-  correlationId: string;
-}): Promise<[Tenant, CreateEvent<TenantEvent>?]> {
+}): Promise<{ tenantKind: TenantKind; tenantKindHasBeenUpdated: boolean }> {
   const tenantKind = await getTenantKindLoadingCertifiedAttributes(
     readModelService,
     tenant.attributes,
     tenant.externalId
   );
-
-  const updatedTenant = {
-    ...tenant,
-    kind: tenantKind,
-  };
-
-  if (tenant.kind !== tenantKind) {
-    const tenantKindUpdatedEvent = toCreateEventTenantKindUpdated(
-      version,
-      tenantKind,
-      updatedTenant,
-      correlationId
-    );
-
-    return [updatedTenant, tenantKindUpdatedEvent];
-  }
-  return [updatedTenant];
+  return tenant.kind !== tenantKind
+    ? { tenantKind, tenantKindHasBeenUpdated: true }
+    : { tenantKind, tenantKindHasBeenUpdated: false };
```

Comment:
```suggestion
  const tenantKindHasBeenUpdated = tenant.kind !== tenantKind
  return { tenantKind, tenantKindHasBeenUpdated} ;
```

Reply:
This suggestion could be consider obsolete, see [here](https://github.com/pagopa/interop-be-monorepo/pull/842#discussion_r1698723943)